build/dind-test/base/Dockerfile

```diff
@@ -0,0 +1,76 @@
+# Copyright 2017 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+FROM gcr.io/google-containers/debian-base-amd64:0.3
+MAINTAINER qlee@google.com
+
+ENV container docker
+
+ARG DEBIAN_FRONTEND=noninteractive
+RUN clean-install apt-utils \
+    apt-transport-https \
+    ca-certificates \
+    curl \
+    gnupg2 \
+    software-properties-common \
+    curl \
+    lxc \
+    iptables \
+    iproute2 \
+    socat \
+    util-linux \
+    mount ebtables \
+    ethtool \
+    vim \
+    bash \
+    procps \
+    less \
+    kmod \
+    systemd \
+    systemd-sysv \
+    jq
+
+# Install docker.
+RUN curl -fsSL https://download.docker.com/linux/debian/gpg | apt-key add - && apt-key fingerprint 0EBFCD88 && add-apt-repository \
+    "deb [arch=amd64] https://download.docker.com/linux/debian \
+    jessie \
+    stable"
+RUN clean-install docker-ce
+
+# Install the magic wrapper.
+COPY dind /bin/dind
+COPY dockerd-entrypoint.sh /bin/dockerd-entrypoint.sh
+
+# Remove unwanted systemd services.
+RUN for i in /lib/systemd/system/sysinit.target.wants/*; do [ "${i##*/}" = "systemd-tmpfiles-setup.service" ] || rm -f "$i"; done; \
```

Comment:
Does `RUN` automatically blow up if something exits 1? or do we need `&&`

Reply:
I believe run means `sh -c "<cmd>"`, and not `sh -e -c "<cmd>"`,  or anything like that.
However, I don't think it matter so much here – we are trimming a well-known set here.
Although, it maybe nicer to be more explicit – backup wanted file firs, remove directory and then re-create it with just that one file.